specification/apimanagement/control-plane/Microsoft.ApiManagement/preview/2017-03-01/apimreports.json

```diff
@@ -0,0 +1,652 @@
+  ],
+  "securityDefinitions": {
+    "apim_key": {
+      "type": "apiKey",
+      "name": "Authorization",
+      "in": "header"
+    }
+  },
+  "security": [
+    {
+      "apim_key": []
+    }
+  ],
+  "paths": {
+    "/reports/byApiId": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Reports_ListByApi",
+        "description": "Lists report records by API.",
+        "x-ms-examples": {
+          "ApiManagementGetReportsByApi": {
+            "$ref": "./examples/ApiManagementGetReportsByApi.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists a collection of Report record.",
+            "schema": {
+              "$ref": "#/definitions/ReportCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/ReportRecordContract"
+      }
+    },
+    "/reports/byUserId": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Reports_ListByUser",
+        "description": "Lists report records by User.",
+        "x-ms-examples": {
+          "ApiManagementGetReportsByUser": {
+            "$ref": "./examples/ApiManagementGetReportsByUser.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "$filter",
+            "in": "query",
+            "required": true,
+            "type": "string",
+            "description": "The filter to apply on the operation."
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/TopQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/SkipQueryParameter"
+          },
+          {
+            "$ref": "./apimanagement.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Lists a collection of Report record.",
+            "schema": {
+              "$ref": "#/definitions/ReportCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        },
+        "x-ms-odata": "#/definitions/ReportRecordContract"
+      }
+    },
+    "/reports/byOperationId": {
+      "get": {
+        "tags": [
+          "Reports"
+        ],
+        "operationId": "Reports_ListByOperation",
```

Comment:
>Reports_ListByOperation [](start = 24, length = 23)

Does this have an operation id parameter?

Reply:
It lists records by api operation. look at example ApiManagementGetReportsByOperation

---
In reply to: [163390885](https://github.com/Azure/azure-rest-api-specs/pull/2269#discussion_r163390885) [](ancestors = 163390885)